concent_api/core/subtask_helpers.py

```diff
@@ -33,6 +33,15 @@ def update_timed_out_subtasks(
         next_deadline__lte      = timezone.now()
     )
 
+    for subtask in clients_subtask_list:  # create locks in database for all subtasks of client
+        try:
+            Subtask.objects.select_for_update().get(pk=subtask.pk)
```

Comment:
Why not locking them in the query above ?

Reply:
I though't about this, but this query is already pretty complicated and I decided to do it like this, locking is more distinct. But I also hesitated in this place, if you think it would be better to do it in query I will

Reply:
It would be better to do this in single query. Good formatting will make it less complicated ;)

Reply:
We want to minimize the number of queries as much as possible. And especially we don't want to make n queries to fetch n objects. We want only one query to fetch them all unless there are reasons preventing us from doing this (e.g. the need to commit a transaction - which is not the case here).

I agree with @rwrzesien, the lock should be in the query above, no matter how complex it is.

Reply:
Agree, done :)